### PR TITLE
Sk reservations

### PR DESCRIPTION
**Pull request by stevkim**
Added dependencies and some other boilerplate code.

*** Comments were left in for testing purposes and clarity for other developers.

Test endpoints connections to the front-end.

### CODE REVIEW

File: server/lib/googleapi.js
Comment (clee4037) [inline server/lib/googleapi.js:1]: Good overview of inputs and outputs of api in comments!

**Review by clee4037 — APPROVED** [review]
Great pseudo coding -- have a clear understanding of each controller function.

File: package.json [hunk 1]
```diff
@@ -4,13 +4,19 @@
...
     "@supabase/supabase-js": "^2.39.1",
     "dotenv": "^16.3.1",
-    "express": "^4.18.2"
+    "express": "^4.18.2",
+    "nodemon": "^3.0.2"
+  },
+  "devDependencies": {
+    "morgan": "^1.10.0"
   }
 }
```
Comment (justcord10) [inline package.json:22]: dependencies look good

File: server/app.js [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
+app.use('/geocode', geocodeRouter);
+// /reservations -> gets you all garages that matches city/zip (need city and/or state in address object)
+// /reservations/test -> gets you all the reservations (need garageId)
+app.use('/reservations', reservationRouter);
 
 app.listen(PORT, () => {
   console.log(`Listening on PORT:${PORT}`);
-})
\ No newline at end of file
+});
```
Comment (justcord10) [inline server/app.js:20]: good routes setup

File: server/controllers/reservationController.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+    if (error) throw error;
+
+    return await data;
+  } catch (error) {
+    console.log(error);
+    return null;
+  }
+}
+
```
Comment (justcord10) [inline server/controllers/reservationController.js:19]: get all garages by zip or city looks good

File: server/controllers/reservationController.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    .eq('id', garageId);
+
+    if (error) throw error;
+
+    return await data;
+  } catch (error) {
+    console.log(error);
+    return null;
+  }
+}
```
Comment (justcord10) [inline server/controllers/reservationController.js:73]: test controller to see all reservations for a garage

File: server/lib/googleapi.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+const getClosestLocation = (origin, targets) => {
+  // origin is the target location for the client
+  // targets are a list of address in our db
+    // each share either a zipcode or city with the origin location
+  // use the google api to get the distances of each target location
+  // use array reduce to find and return the location thats closest
+}
+
+module.exports = { createGeocode };
```
Comment (justcord10) [inline server/lib/googleapi.js:36]: function to get lat and lon from the Google Maps API

**Review by justcord10 — APPROVED** [review]
Looks good, left some comments sprinkled throughout summarizing what I think was going on but overall looks good. clean code

**Pull request merged**